src/main/java/net/minecraftforge/client/model/ModelDynBucket.java

```diff
@@ -0,0 +1,243 @@
+package net.minecraftforge.client.model;
+
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Maps;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.block.model.BakedQuad;
+import net.minecraft.client.renderer.block.model.ItemCameraTransforms;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.renderer.vertex.VertexFormat;
+import net.minecraft.client.resources.IResourceManager;
+import net.minecraft.client.resources.model.IBakedModel;
+import net.minecraft.client.resources.model.ModelResourceLocation;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.client.model.pipeline.UnpackedBakedQuad;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidContainerRegistry;
+import net.minecraftforge.fluids.FluidRegistry;
+import net.minecraftforge.fluids.FluidStack;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import javax.vecmath.Matrix4f;
+
+public class ModelDynBucket implements IModel {
+  public static final ModelResourceLocation LOCATION = new ModelResourceLocation(new ResourceLocation("forge", "dynbucket"), "inventory");
+
+  private static final float NORTH_Z_FLUID = 7.49f / 16f;
+  private static final float SOUTH_Z_FLUID = 8.51f / 16f;
+
+  private static final float NORTH_Z_COVER = 7.48f / 16f;
+  private static final float SOUTH_Z_COVER = 8.52f / 16f;
+
+  protected final IModel bucket;
+  protected static final ResourceLocation interior = new ResourceLocation("forge", "items/bucket_interior");
+  protected static final ResourceLocation front = new ResourceLocation("forge", "items/bucket_front");
+
+  private boolean hasTextures;
+
+  public ModelDynBucket(IModel bucket) {
+    this.bucket = bucket;
+  }
+
+  @Override
+  public Collection<ResourceLocation> getDependencies() {
+    return bucket.getDependencies();
+  }
+
+  @Override
+  public Collection<ResourceLocation> getTextures() {
+    ImmutableSet.Builder<ResourceLocation> builder = ImmutableSet.builder();
+
+    // Do the textures for interiors exist?
+
+    try {
+      ResourceLocation loc = interior;
+      loc = new ResourceLocation(loc.getResourceDomain(), "textures/" + loc.getResourcePath() + ".png");
```

Comment:
IRetexturableModel is probably better than hardcoded locations, that way, for example, this model can be reused for IC2 cans, or something similar.